File: C++/Merge two BST 's.cpp
class Solution {
  public:
        vector<int> inorder1(Node* root1, vector<int> &res1){
            if(root1==NULL) return res1;
                inorder1(root1->left, res1);
                res1.push_back(root1->data);
                inorder1(root1->right, res1);
            return res1;
        }
        vector<int> merge(vector<int>res1, vector<int>res2, vector<int> & res){
            int m=res1.size();
            int n=res2.size();
            int i=0, j=0;
            while(i<m and j<n){
                if(res1[i]<=res2[j]){
                    res.push_back(res1[i]);
                    i++;
                }else{
                    res.push_back(res2[j]);
                    j++;
                }
            }
            while(i<m){
                res.push_back(res1[i]);
                i++;
            }
            while(j<n){
                res.push_back(res2[j]);
                j++;
            }
            return res;
        }
    vector<int> merge(Node *root1, Node *root2) {
        vector<int>res2;
        vector<int>res1;
        vector<int>res;
        inorder1(root1, res1);
        inorder1(root2, res2);
        res=merge(res1, res2, res);
        return res;
    }
};

File: JAVA/Merge two BST 's.java
class Solution {
    ArrayList<Integer>list1;
    ArrayList<Integer>list2;
    public List<Integer> merge(Node root1, Node root2) {
        list1=new ArrayList<>();
        list2=new ArrayList<>();
        inorder(root1,list1);
        inorder(root2,list2);
        int i=0,j=0;
        ArrayList<Integer>res=new ArrayList<>();
        while(i<list1.size()&&j<list2.size()){
            if(list1.get(i)<=list2.get(j)){
                res.add(list1.get(i));
                i++;
            }else{
                res.add(list2.get(j));
                j++;
            }
        }
        while(i<list1.size()){
            res.add(list1.get(i));
            i++;
        }
        while(j<list2.size()){
            res.add(list2.get(j));
            j++;
        }
        return res;
    }
    public void inorder(Node root,ArrayList<Integer>list){
        if(root==null) return;
        inorder(root.left,list);
        list.add(root.data);
        inorder(root.right,list);
    }
}
